src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParametersTests.cs

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.ConsiderPassingBaseTypesAsParameters,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
+{
+    public class ConsiderPassingBaseTypesAsParametersTests
+    {
+        [Fact]
+        public async Task Unused_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class A {}
+public class B : A {}
+
+public class Z
+{
+    public void M(B b)
+    {
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MethodCall_TypeWithBase()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class A
+{
+    public int GetSomething() => 42;
+}
+
+public class B : A
+{
+    public int GetSomethingElse() => 42;
+}
+
+public class Z
+{
+    public void UsesOnlyBaseType(B {|#0:b|})
+    {
+        b.GetSomething();
+    }
```

Comment:
Can we get to a scenario where changing `B` to `A` can cause `GetSomething` to be statically (at compile-time) or dynamically (at run-time) bound to something different?

Reply:
I can't think of anything.